Source/CarthageKit/Xcode.swift

```diff
@@ -913,6 +917,47 @@ public func buildDependencyProject(dependency: ProjectIdentifier, rootDirectoryU
 		}
 }
 
+public func signingIdentities() -> SignalProducer<String, CarthageError> {
+	let securityTask = TaskDescription(launchPath: "/usr/bin/security", arguments: ["find-identity", "-v", "-p", "codesigning"])
+	
+	return launchTask(securityTask)
+		|> ignoreTaskData
+		|> mapError { CarthageError.TaskError($0) }
+		|> map { (data: NSData) -> String in
+			return NSString(data: data, encoding: NSStringEncoding(NSUTF8StringEncoding))! as String
+		}
+		|> flatMap(.Merge) { (string: String) -> SignalProducer<String, CarthageError> in
+			return string.linesProducer |> promoteErrors(CarthageError.self)
+		}
+		|> map { (identityLine: String) -> String? in
+			var error: NSError? = nil
+			let regex = NSRegularExpression(pattern: "\".+\"", options: nil, error: &error)
+			if let match = regex?.firstMatchInString(identityLine, options: nil, range: NSRange(location: 0, length: count(identityLine))) as NSTextCheckingResult? {
+				let quotedIdentityName = (identityLine as NSString).substringWithRange(match.range)
+				let identityName = quotedIdentityName.stringByTrimmingCharactersInSet(NSCharacterSet(charactersInString: "\""))
```

Comment:
If you use the capturing group in the regex like I suggested above, you can get rid of most of this and use `let identityName = (identityLine as NSString).substringWithRange(match.rangeAtIndex(1))`.